src/qibo/models/circuit.py

```diff
@@ -1099,14 +1099,20 @@ def execute(self, initial_state=None, nshots=1000):
             self._final_state = self.compiled.result(state, nshots)
             return self._final_state
         else:
-            from qibo.backends import GlobalBackend
+            from qibo.backends import _Global
+
+            _Global.resolve_global()
+            transpiler = _Global.get_transpiler()
+            transpiled_circuit, _ = transpiler(self)  # pylint: disable=E1102
 
             if self.accelerators:  # pragma: no cover
-                return GlobalBackend().execute_distributed_circuit(
-                    self, initial_state, nshots
+                return _Global.get_backend().execute_distributed_circuit(
+                    transpiled_circuit, initial_state, nshots
                 )
             else:
-                return GlobalBackend().execute_circuit(self, initial_state, nshots)
+                return _Global.get_backend().execute_circuit(
```

Comment:
```suggestion
                return _Global.backend().execute_circuit(
```

Reply:
Since you're getting the backend in both branches, you could factor out this instruction before branching:
```py
backend = _Global.backend()
if ...:
    return backend.exec...
return backend.exec...
```

Reply:
Updated it. #U2